control_plane/src/pageserver.rs

```diff
@@ -150,48 +151,39 @@ impl PageServerNode {
             init_config_overrides.push("auth_validation_public_key_path='auth_public_key.pem'");
         }
 
-        let mut pageserver_process = self
+        let pageserver_process = self
             .start_node(&init_config_overrides, &self.env.base_data_dir, true)
             .with_context(|| {
                 format!(
                     "Failed to start a process for pageserver {}",
                     self.env.pageserver.id,
                 )
             })?;
+        let pageserver_process = scopeguard::guard(pageserver_process, |mut pageserver_process| {
+            // stop_process() assumes that someone is wait()ing for the child process to exit.
+            // For the regular case (neon_local start), that's PID 1 because `neon_local start` exits
+            // after it has started the pageserver.
+            // But for initialize(), we created the process and we're still alive.
+            // So, kick of a wait() thread so that stop_process() so that eventually
+            // stop_process() will observe ESRCH when it (repeatedly) tries to `kill` the process.
+            let wait_thread = std::thread::spawn(move || pageserver_process.wait());
+            match stop_process(false, "pageserver", &self.pid_file()) {
+                Ok(()) => {
+                    let st = wait_thread.join();
+                    println!("Stopped pageserver process; status: {st:?}")
+                }
+                Err(e) => eprintln!("Failed to stop pageserver process: {e:#}"),
+            }
+        });
```

Comment:
Seems a bit pointless to do both `Child::wait()` and the polling in `stop_process` that waits for the process to be gone. `Child::wait()` is enough.